core/src/main/java/com/datastax/oss/driver/api/core/metadata/schema/FunctionSignature.java

```diff
@@ -95,4 +95,19 @@ public boolean equals(Object other) {
   public int hashCode() {
     return Objects.hash(name, parameterTypes);
   }
+
+  @Override
+  public String toString() {
+    StringBuilder builder = new StringBuilder(name.asInternal()).append('(');
+    boolean first = true;
+    for (DataType type : parameterTypes) {
+      if (first) {
+        first = false;
+      } else {
+        builder.append(", ");
+      }
+      builder.append(type.asCql(true, true));
```

Comment:
we are sure that type is always non-null?

Reply:
Yes. The nullability annotations are not powerful enough to express nullability of collection elements, but in practice function parameters always have types.